File: ReferenceImplementation/ReferenceProject.backend/src/ReferenceProject/backend/models/ComplaintStatus.java
package ReferenceProject.backend.models;

import javax.xml.bind.annotation.XmlEnumValue;

public enum ComplaintStatus {
	
	@XmlEnumValue("0") VALUE0("User is filling out complaint"),
	@XmlEnumValue("1") VALUE1("Complaint is sent to administration"),
	@XmlEnumValue("2") VALUE2("Complaint is in process"),
	@XmlEnumValue("3") VALUE3("Complaint has been handled by the administration");
	
	private String value;
	
	ComplaintStatus(String value) {
		this.value = value;
	}
	
	public String getValue() {
		return this.value;
	}
	
}

File: ReferenceImplementation/Citizenapp/bundles/md2_models/models/ComplaintStatus.js
define([
    "dojo/_base/declare",
    "md2_runtime/datatypes/_Enum"
],
function(declare, _Enum) {
    
    var ComplaintStatus = declare([_Enum], {
        
        _datatype: "ComplaintStatus",
        
        _enum: {
            VALUE0: "User is filling out complaint",
            VALUE1: "Complaint is sent to administration",
            VALUE2: "Complaint is in process",
            VALUE3: "Complaint has been handled by the administration"
        }
        
    });
    
    /**
     * Enum Factory
     */
    return declare([], {
        
        datatype: "ComplaintStatus",
        
        create: function(value) {
            return new ComplaintStatus(value, this.typeFactory);
        }
        
    });
});
